src/quick-lint-js/parse.h

```diff
@@ -1549,6 +1555,9 @@ class parser {
         method_attributes = function_attributes::async_generator;
         this->skip();
       }
+      if (this->peek().type == token_type::kw_static) {
+        async_static = true;
```

Comment:
I'd report the error here, and highlight the `async static` keywords instead of the method name. I think that'd simplify the logic and also be a better user experience.

Reply:
That's what I tried originally, but then it triggers a false positive for an async method named static. (`async static() {}`)

Perhaps I should use the transaction stuff and just check for a left paren right then and there?

Reply:
> That's what I tried originally, but then it triggers a false positive for an async method named static. (`async static() {}`)

I realized that soon after I made my comment. =X